src/options/settings/components/SearchInjection.js

```diff
@@ -4,20 +4,24 @@ import PropTypes from 'prop-types'
 import Checkbox from './Checkbox'
 import styles from './SearchInjection.css'
 
-const SearchInjection = ({ isInjectionEnabled, toggleInjection }) => {
+const SearchInjection = ({ injectionPreference, toggleGoogle, toggleDDG }) => {
```

Comment:
One thing to note with this is that now you're passing in separate state changing callbacks for each `<Checkbox>` but their states are inside the single `injectionPreference` object prop. Bit inconsistent and another thing is that every time we want to add another `<Checkbox>` we've got to change the signature of this view.

If I was to write it now (not saying "you should have done it this way!"; things like this are always more obvious in hindsight or looking at others' code) I would probably just have this view accept a single `children` array of `React.Node`s prop, so it becomes like a "wrapper view", and then in your rendering code (in `SearchInjectionContainer`) connect the baby views and pass them down like:

```js
<SearchInjectionSettings>
  <Checkbox isChecked={...} handleChange={...}>Google</Checkbox>
  <Checkbox isChecked={...} handleChange={...}>DuckDuckGo</Checkbox>
</SearchInjectionSettings>
```

Then you don't need to worry about passing around so many props, having to defining them everywhere, and need to remember about all of them when you make some changes, etc. The base idea is taking a architecture like "container -> view -> ... -> view" and trying reducing the depth to a single level "container -> views" (and split out into multiple containers with separate concerns if it becomes big/complicated).

You can change it if you want, but really not a high-prio for this work :) more of something to think about in future UI coding (I also need to try and get into this thinking habit with my UI coding!)

Reply:
Oh yeah looking at it now, makes me realise it wasn't the best way. Thanks for the suggestion, I will try to change it right now!